src/SixLabors.Fonts/Tables/AdvancedTypographic/GSubTable.cs

```diff
@@ -0,0 +1,152 @@
+            // +==========+=========================+===============================================================================+
+            // | uint16   | majorVersion            | Major version of the GSUB table, = 1                                          |
+            // +----------+-------------------------+-------------------------------------------------------------------------------+
+            // | uint16   | minorVersion            | Minor version of the GSUB table, = 1                                          |
+            // +----------+-------------------------+-------------------------------------------------------------------------------+
+            // | Offset16 | scriptListOffset        | Offset to ScriptList table, from beginning of GSUB table                      |
+            // +----------+-------------------------+-------------------------------------------------------------------------------+
+            // | Offset16 | featureListOffset       | Offset to FeatureList table, from beginning of GSUB table                     |
+            // +----------+-------------------------+-------------------------------------------------------------------------------+
+            // | Offset16 | lookupListOffset        | Offset to LookupList table, from beginning of GSUB table                      |
+            // +----------+-------------------------+-------------------------------------------------------------------------------+
+            // | Offset32 | featureVariationsOffset | Offset to FeatureVariations table, from beginning of GSUB table (may be NULL) |
+            // +----------+-------------------------+-------------------------------------------------------------------------------+
+            ushort majorVersion = reader.ReadUInt16();
+            ushort minorVersion = reader.ReadUInt16();
+
+            ushort scriptListOffset = reader.ReadOffset16();
+            ushort featureListOffset = reader.ReadOffset16();
+            ushort lookupListOffset = reader.ReadOffset16();
+            uint featureVariationsOffset = (minorVersion == 1) ? reader.ReadOffset32() : 0;
+
+            // TODO: Optimization. Allow only reading the scriptList.
+            var scriptList = ScriptList.Load(reader, scriptListOffset);
+
+            var featureList = FeatureListTable.Load(reader, featureListOffset);
+
+            var lookupList = LookupListTable.Load(reader, lookupListOffset);
+
+            // TODO: Feature Variations.
+            return new GSubTable(scriptList, featureList, lookupList);
+        }
+
+        public void ApplySubstitions(GlyphSubstitutionCollection collection, ushort index, int count)
+        {
+            collection.GetCodePointAndGlyphIds(index, out CodePoint codePoint, out int _, out IEnumerable<int> _);
+
+            ScriptListTable scriptListTable = this.ScriptList.Default();
+            Tag[] tags = UnicodeScriptTagMap.Instance[CodePoint.GetScript(codePoint)];
```

Comment:
I noticed, that the `CodePoint.GetScript` method returns a wrong value for arabic characters. For example "ب", CodePoint 1576 (hex: 628) it returns `CaucasianAlbanian`. My understanding is, that this is because `UnicodeCategory.trie` was build with only latin characters in mind. Is that correct @JimBobSquarePants ?

Reply:
Hmmm no, must be a bug in my generation: There should be 1291 Arabic codepoints in there according to the input txt file it’s generated from. Will double check

Reply:
Fixed. Generator was fine I was just reading the wrong trie!! 🤣 I've added a few test ranges to confirm it's correct now for now but will flesh out in the future.